holoviews/operation/datashader.py

```diff
@@ -202,7 +209,9 @@ def _get_sampling(self, element, x, y, ndim=2, default=None):
         xs, ys = (np.linspace(xstart+xunit/2., xend-xunit/2., width),
                   np.linspace(ystart+yunit/2., yend-yunit/2., height))
 
-        return ((xstart, xend), (ystart, yend)), (xs, ys), (width, height), (xtype, ytype)
+        width *= self.p.pixel_ratio
+        height *= self.p.pixel_ratio
+        return ((xstart, xend), (ystart, yend)), (xs, ys), (int(width), int(height)), (xtype, ytype)
```

Comment:
Up to you on style, but my own vote would be to apply int() in the same operation as applying the ratio, so that the height and width datatype doesn't briefly differ from the expected integer type.

Reply:
Addressed in ca7935045